### PR TITLE
add watermark_strategy_expression

### PR DESCRIPTION
**Pull request by st1page**
refers FlinkSQL's watermark concept, change the watermark's `time_out` to `watermark_strategy_expression`. The PR
will be merged into #2 after being approved

### CODE REVIEW

**Comment by TennyZhuang** [comment]
#2

**Comment by yuhao-su** [comment]
LGTM

File: rfcs/0002-watermark-filter.md [hunk 1]
```diff
@@ -172,9 +172,12 @@ The core idea of Exchange is **Prop 1.1.** We need also to align the watermarks,
 We can discuss the syntax later, but now we can implement that as a table function and expose the watermarkFilter’s behavior directly.
 
 ```plain
-WATERMARK(orders, time_column, timeout: interval)
+WATERMARK(orders, time_column, watermark_strategy_expression)
 ```
 
+And the concept `watermark_strategy_expression` refers [FlinkSQL](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/table/sql/create/#watermark). The expression is evaluated for every record and update the watermark if the result greater than the current watermark.
+The new synatx is more friendly for flinkSQL user and offers user methods defining their strategy
```
Comment (BugenZhao) [inline rfcs/0002-watermark-filter.md:179]:
```suggestion
The new syntax is more friendly for flinkSQL user and offers user methods defining their strategy.
```

File: rfcs/0002-watermark-filter.md [hunk 2]
```diff
@@ -235,6 +238,52 @@ WINDOW w AS (
 )
 ```
 
+And `watermark_strategy_expression` offers potential to make user define their own strategy. Some following case is fancy and not very well-defined. You can treat them as a pseudocode which we might implement in furture
```
Comment (BugenZhao) [inline rfcs/0002-watermark-filter.md:241]:
```suggestion
And `watermark_strategy_expression` offers potential to make user define their own strategy. Some following case is fancy and not very well-defined. You can treat them as a pseudocode which we might implement in future.
```

**Pull request merged**